src/model/analysis.ts

```diff
@@ -118,6 +122,33 @@ export class Analysis implements Queryable {
     return Array.from(result);
   }
 
+  /**
+   * Potentially narrow down the document that contains the sourceRange.
+   * For example, if a source range is inside a inlineDocument, this function
+   * will narrow down the document to the most specific inline document.
+   *
+   * @param sourceRange Source range to search for in a document
+   * @param document The document that contains the source range
+   */
+  getDocumentContaining(sourceRange: SourceRange|undefined, document: Document):
```

Comment:
`document` seems unnecessary here if we have the analysis and the source range.

Could be written as:

```typescript
getDocumentContaining(sourceRange: SourceRange|undefined) {
  if (!sourceRange) {
    return undefined;
  }
  let mostSpecificDocument: undefined|Document = undefined;
  for (const doc of this.getFeatures({kind: 'document', id: sourceRange.file})) {
    ...
  }
}
```

Reply:
I have updated the code, but I couldn't do the `id: sourceRange.file`, as inline documents explicitly do not have the url as an identifier (https://github.com/Polymer/polymer-analyzer/blob/master/src/model/document.ts#L135) I am not sure in what other way to retrieve this information. Maybe a recursive search starting at the root document would work?

Reply:
Huh, I see. What about:

```javascript
const [outerDoc] = this.getFeatures({kind: 'document', id: sourceRange.file}));
if (outerDoc === undefined) {
  return undefined;
}
for (const doc of outerDoc.getFeatures({kind: 'document'})) {
   ...
}
```